application/controllers/AuthenticationController.php

```diff
@@ -35,11 +41,32 @@ public function loginAction()
             $this->redirectNow(Url::fromPath('setup'));
         }
         $form = new LoginForm();
+        if (RememberMe::hasCookie()) {
+            $authenticated = false;
+            try {
+                $rememberMe = RememberMe::fromCookie();
+                $authenticated = $rememberMe->authenticate();
+                if ($authenticated) {
```

Comment:
The authenticated check can be moved outside the if. You already have a `! $authenticated` check there. Also the line can be shortened to `$authenticated = RememberMe::fromCookie()->authenticate()`.